src/cache.c

```diff
@@ -9,6 +9,12 @@ cache_init(Cache *cache)
 		return;
 	}
 
+	/*
+	 * Make sure that the cache object was allocated in its own context This
+	 * is needed to make sure cache_destroy doesn't need to pfree(cache) etc.
+	 */
+	Assert(MemoryContextContains(cache_memory_ctx(cache), cache));
```

Comment:
Ok, so I understand this assert is not related  to segfault. But, still, the comment seems to indicate this check "is needed", so why an assert? If this is a debug thing I think the comment is a bit misleading.